lib/domain/blocs/form_bloc/form_bloc.dart

```
@@ -0,0 +1,71 @@
+import 'package:c2s/data/json_data/get_entry_response_data.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import '../../repositories/abstract_entries_repo.dart';
+import 'form_event.dart';
+import 'form_state.dart';
+
+class FormBloc extends Bloc<FormEvent, FormState> {
+  final AbstractEntriesRepo repository;
+
+  FormBloc(this.repository) : super(FormInitial()) {
+    // Handle LoadEntryEvent
+    on<LoadEntryEvent>((event, emit) async {
+      emit(FormLoading());
+      if (event.id != 'empty') {
+        try {
+          final entryData = await repository.getEntry(event.token, event.id);
+          entryData.data?.date = entryData.data?.date
+              ?.substring(0, entryData.data?.date?.indexOf('T'));
+          emit(FormLoaded(entryData: entryData));
+        } catch (e) {
+          emit(FormError(e.toString()));
+        }
+      } else {
```

if there isn't an id for the entry to fetch it's details, you should raise an error not emitting null entry response

---

no this is for the first form in case it is first opened it doesn't give an error